File: appli-android/app/src/main/java/com/example/sportapp/core/network/CycleWorkoutApi.kt
package com.example.sportapp.core.network

import com.example.sportapp.core.data.model.CycleWorkout
import retrofit2.http.*

interface CycleWorkoutApi {

    @GET("cycle-workouts")
    suspend fun getAll(): List<CycleWorkout>

    @PUT("cycle-workouts/{uuid}")
    suspend fun upsert(@Path("uuid") uuid: String, @Body cycleWorkout: CycleWorkout)

    @PUT("cycle-workouts/bulk")
    suspend fun upsertAll(@Body cycleWorkouts: List<CycleWorkout>)

    @DELETE("cycle-workouts/{uuid}")
    suspend fun delete(@Path("uuid") uuid: String)
}

File: appli-web/src/app/core/api/cycle-workout-api.ts
import { HttpClient } from '@angular/common/http';
import { Injectable, inject } from '@angular/core';
import { Observable } from 'rxjs';
import { CycleWorkout } from '@core/models/cycle-workout.model';
import { API_BASE_URL } from './api.config';

@Injectable({ providedIn: 'root' })
export class CycleWorkoutApi {
  private readonly http = inject(HttpClient);
  private readonly base = `${API_BASE_URL}/cycle-workouts`;

  getAll(): Observable<CycleWorkout[]> {
    return this.http.get<CycleWorkout[]>(this.base);
  }

  upsert(c: CycleWorkout): Observable<CycleWorkout> {
    // Corps wire attendu par CycleWorkoutCreate (camelCase).
    return this.http.put<CycleWorkout>(`${this.base}/${c.uuid}`, {
      uuid: c.uuid,
      trainingCycleUUID: c.trainingCycleUUID,
      plannedWorkoutUUID: c.plannedWorkoutUUID,
      updatedAt: c.updatedAt,
    });
  }

  /** Upsert groupé `PUT /cycle-workouts/bulk` (miroir upsertAll Android) — 1 requête pour N rows. */
  upsertAll(items: CycleWorkout[]): Observable<CycleWorkout[]> {
    return this.http.put<CycleWorkout[]>(
      `${this.base}/bulk`,
      items.map((c) => ({
        uuid: c.uuid,
        trainingCycleUUID: c.trainingCycleUUID,
        plannedWorkoutUUID: c.plannedWorkoutUUID,
        updatedAt: c.updatedAt,
      })),
    );
  }

  delete(uuid: string): Observable<unknown> {
    return this.http.delete(`${this.base}/${uuid}`);
  }
}
